pkg/synthetictests/kubelet.go

```diff
@@ -280,7 +280,7 @@ func testNodeUpgradeTransitions(events monitorapi.Intervals, kubeClientConfig *r
 			}
 			// If events indicate that a node reboot reason happened more than once due to machine config update AND the node kernel is a realtime kernel
 			// We increase the count to assess if we should skip
-			if _, ok := realtimeNodeConfigReboots[node]; ok && strings.Contains(event.Message, "reason/Reboot roles/worker Node will reboot into config ") {
+			if _, ok := realtimeNodeConfigReboots[node]; ok && strings.Contains(event.Message, "Update completed for config rendered-worker-") {
```

Comment:
Looking at MCO daemon.go, I'm not convinced this is accurate, but I don't know MCO well enough to know that this event is a good indicator for counting reboots.

Reply:
I'll leave the /approve but get someone from MCO to do the lgtm please

Reply:
Looking at the event generated at https://github.com/openshift/machine-config-operator/blob/b52e75eabe58c8e13edf3948dc9509ea9f93b3d9/pkg/daemon/daemon.go#L1686 i.e. `Update completed for config ...` would be appropriate indication that MCO has finished applying update. MCO skips reboot for few cases but I think this code is more about applying RT kernel, so it would be fine to assume that MCO will reboot when it applies the config and this event will be triggered.

Just a note that this event log message may change in future if there is a genuine reason. In that case, this will require an update.

Reply:
One question, we are adding "rendered-worker-" in the search string, is this always going to be a worker node or it can be master/custom pool as well?